File: util/src/entities/Team.ts
import { Column, Entity, JoinColumn, ManyToMany, ManyToOne, RelationId } from "typeorm";
import { BaseClass } from "./BaseClass";
import { TeamMember } from "./TeamMember";
import { User } from "./User";

@Entity("teams")
export class Team extends BaseClass {
	@Column()
	icon?: string;

	@RelationId((team: Team) => team.members)
	member_ids: string[];

	@JoinColumn({ name: "member_ids" })
	@ManyToMany(() => TeamMember, (member: TeamMember) => member.id)
	members: TeamMember[];

	@Column()
	name: string;

	@RelationId((team: Team) => team.owner_user)
	owner_user_id: string;

	@JoinColumn({ name: "owner_user_id" })
	@ManyToOne(() => User, (user: User) => user.id)
	owner_user: User;
}

File: extra/admin-api/Spacebar.Db/Models/Team.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Spacebar.Db.Models;

[Table("teams")]
public partial class Team
{
    [Key]
    [Column("id", TypeName = "character varying")]
    public string Id { get; set; } = null!;

    [Column("icon", TypeName = "character varying")]
    public string? Icon { get; set; }

    [Column("name", TypeName = "character varying")]
    public string Name { get; set; } = null!;

    [Column("owner_user_id", TypeName = "character varying")]
    public string? OwnerUserId { get; set; }

    [InverseProperty("Team")]
    public virtual ICollection<Application> Applications { get; set; } = new List<Application>();

    [ForeignKey("OwnerUserId")]
    [InverseProperty("Teams")]
    public virtual User? OwnerUser { get; set; }

    [InverseProperty("Team")]
    public virtual ICollection<TeamMember> TeamMembers { get; set; } = new List<TeamMember>();
}
